feat(sql-review): new rule for checking no position in add column clause

whhe opened the pull request:
Statements like `alter table t add column a ... after b` are offline ddl on some 4.x versions of OceanBase (MySQL mode) and online ddl for others, so there should be a warning for it to help the DBA filter it out.

For localization, I just add the rule to zh-CN.json and en-US.json, as other languages ​​are too difficult for me.

d-bytebase commented:
We are missing i18n for other locales.

@rebelice can help with the review, and thanks for the contribution!

ecmadao commented:
> We are missing i18n for other locales.

i18n test is broken, fixed here https://github.com/bytebase/bytebase/pull/11973

whhe commented:
> > We are missing i18n for other locales.
> 
> i18n test is broken, fixed here #11973

I rebased this branch and the frontend-tests failed. Is it due to missing i18n text?

whhe commented:
Missing i18n text added. I used chatgpt for translation and I'm not sure if the text is correct.

rebelice reviewed (approved):
Thanks for the contribution!
